module-system/sov-modules-stf-template/src/lib.rs

```diff
@@ -87,30 +104,53 @@ where
             .expect("Storage update must succeed");
     }
 
-    fn begin_slot(&mut self, witness: Self::Witness) {
-        self.checkpoint = Some(StateCheckpoint::with_witness(
-            self.current_storage.clone(),
-            witness,
-        ));
-    }
-
-    fn apply_blob(
+    fn apply_block<'a, I>(
         &mut self,
-        blob: &mut B,
-        _misbehavior_hint: Option<Self::MisbehaviorProof>,
-    ) -> BatchReceipt<Self::BatchReceiptContents, Self::TxReceiptContents> {
-        match self.apply_blob(blob) {
-            Ok(batch) => batch,
-            Err(e) => e.into(),
+        height: u64,
+        witness: Self::Witness,
+        blobs: I,
+    ) -> BlockResult<
+        Self::StateRoot,
+        Self::BatchReceiptContents,
+        Self::TxReceiptContents,
+        Self::Witness,
+    >
+    where
+        I: IntoIterator<Item = &'a mut B>,
+    {
+        self.begin_slot(witness);
+
+        let mut batch_receipts = vec![];
+        for (blob_idx, blob) in blobs.into_iter().enumerate() {
+            let batch_receipt = match self.apply_blob(blob) {
```

Comment:
You can use `?` instead of pattern matching.

Reply:
That's a good idea.

It didn't work originally, because of :

```
124 |               let batch_receipt = self.apply_blob(blob)?;
    |                                                        ^ cannot use the `?` operator in a method that returns `BlockResult<RootHash, SequencerOutcome, TxEffect, <<C as Spec>::Storage as Storage>::Witness>`
    |
    = help: the trait `FromResidual<Result<Infallible, ApplyBatchError>>` is not implemented for `BlockResult<RootHash, SequencerOutcome, TxEffect, <<C as Spec>::Storage as Storage>::Witness>`
    
```

But I can implement needed trait.

Reply:
That's actually interesting observation.

If single blob in slot has failed to be applied, should we abort whole slot? If so, then `?` can be used.

But if slot should be continued, then just simple `.into()` is fine.

Reply:
Actually in current logic, we keep going. @preston-evans98 please confirm if this is desired behaviour.

Also, I've get rid of `match` for something more compact.

Reply:
Yes, we want to keep going. Otherwise, it would be pretty easy to DOS the rollup by posting invalid blobs in every slot